neonvm/apis/neonvm/v1/virtualmachine_types.go

```diff
@@ -196,9 +197,55 @@ type GuestSettings struct {
 	// +optional
 	Sysctl []string `json:"sysctl,omitempty"`
 
-	// Swap controls settings for adding a swap disk to the VM.
+	// Swap adds a swap disk with the provided size.
+	//
+	// If Swap is provided, SwapV2 MUST NOT be provided, and vice versa.
+	//
+	// +optional
+	Swap *resource.Quantity `json:"swap,omitempty"`
+
+	// SwapV2 controls settings for adding a swap disk to the VM.
+	//
+	// SwapV2 is a temporary newer version of the Swap field.
+	//
+	// Eventually, after all VMs have moved from Swap to SwapV2, we can change the type of the Swap
+	// field to SwapInfo, move VMs from SwapV2 back to Swap, and then remove SwapV2.
+	//
+	// More information here: https://neondb.slack.com/archives/C06SW383C79/p1713298689471319
+	//
 	// +optional
-	Swap *SwapInfo `json:"swap,omitempty"`
+	SwapV2 *SwapInfo `json:"swapV2,omitempty"`
```

Comment:
Can be named `SwapInfo`, not SwapV2, so we can keep it indefinitely

Reply:
Good idea. Renamed.